push-image.sh

```diff
@@ -1,76 +1,94 @@
 #!/usr/bin/env bash
-set -e
+set -euo pipefail
 
-# Push the 'conjur' image to various Docker registries
-# Push tagged images on master branch
-# Release images can be created by passing the desired tag to this script
-# Ex: ./push-image 4.9.5.1
+# Pushes the 'conjur' image to various Docker registries
+# There are essentially three flows this script supports
+# - Pushing images to the internal registry during the early pipeline stage
+# - Pushing an "edge"-tagged image to DockerHub
+# - Publishing images to public registries on new project releases during a
+#   tag-triggered build
+#
+# The default flow when no parameters are supplied is to publish images to
+# public registries. This does require that the TAG_NAME variable is set, which
+# happens automatically in tag-triggered builds.
+#
+# To publish to a (private) internal registry, run the script with the "-r" flag
+# to supply the registry prefix.
+#
+# To publish an "edge"-tagged release to DockerHub, run the script with the "-e"
+# flag.
 
 # shellcheck disable=SC1091
-. version_utils.sh
+. build_utils.sh
 
-if [[ -z "${TAG_NAME:-}" ]]; then
-  echo "Please supply environment variable TAG_NAME."
-  echo "If you see this error in Jenkins it means the publish script was run"
-  echo "for a build that wasn't triggered by a tag -" \
-    "please check publish stage conditions."
-  exit 1
-fi
-
-TAG="${1:-$(version_tag)}"
-VERSION="$(< VERSION)"
-SOURCE_IMAGE="conjur:$TAG"
-RH_SOURCE_IMAGE="conjur-ubi:$TAG"
+function print_help() {
+  echo "Usage: $0 [ -r <registry prefix> ] [ -e ]"
+  echo " -r optional registry prefix to prepend to '/cyberark/conjur'"
+  echo " -e optional flag to publish edge-tagged image to DockerHub"
```

Comment:
Can we change these to long-form names instead?
```
--registry-prefix
--edge
```

I would much prefer to see this in the Jenkinsfile:
```
./push-image.sh --edge
```

Reply:
that's a fair point. I'll work that in.